File: c++/binary search.cpp
// c++ program to implement iterative binary search
#include<bits/stdc++.h>
using namespace std;
// an iterative binary search function.
int binarysearch(int arr[],int l, int r,int x)
  {
while(l<=r)
{
int m=1+(r-1)/2:
// check if x is present at mid 
if (arr[m]==x)
  return m;
// if x is greater, ignore left half
if(arr[m]<x)
  l=m+1;
// if x is smaller, ignore right half
else r=m-1;
}
// if we reach here, then element was not present 
return -1;
}
//driver code
int main(void)
{
int arr[]={2,3,4,10,40};
int x=10;
int n= sizeof(arr)/sizeof(arr[0]);
int result= binary search(arr,0,n-1,x);
(result==-1)
  ? cout<<"element is not present in array"
  : cout<<"element is presnt at index"<< result;
return 0;
}

File: Python/binary_search.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1

    while left <= right:
        mid = (left + right) // 2

        if arr[mid] == target:
            return mid  
        elif arr[mid] < target:
            left = mid + 1  
        else:
            right = mid - 1  
    return -1  

arr_str = input("Enter a sorted array of numbers (space-separated): ")
arr = list(map(int, arr_str.split()))

target = int(input("Enter the target element to search for: "))

result = binary_search(arr, target)

if result != -1:
    print(f"Element {target} found at index {result}")
else:
    print(f"Element {target} not found in the array")
